test/scripts/test_private_network.sh

```diff
@@ -35,6 +41,26 @@ ${GOPATH}/bin/goal network stop -r ${NETROOTPATH}
 
 ${GOPATH}/bin/goal network delete -r ${NETROOTPATH}
 
+# Test that genesis generation works correctly
+${GOPATH}/bin/goal network genesis -r ${GENFILESPATH} -t ./test/testdata/nettemplates/TwoNodes50Each.json
```

Comment:
I agree that there is a usability issue here, and it's confusing to not distinguish between the network directory and genesis directory. I've forced the other commands to mark `rootdir` as required (instead of inheriting the global flag), but the disclaimer is that this is uglier as there is a fair bit of code repetition. I also forced the `goal network genesis` command to take `--genesisdir` as a required flag.

Reply:
It looks like -r is not required so you could return an error if it is used. This seems to hide it from the help output:
```
       networkGenesisCmd.SetHelpFunc(func(command *cobra.Command, strings []string) {
               command.Flags().MarkHidden("rootdir")
               command.Parent().HelpFunc()(command, strings)
       })
```